open-craft/frontend-lib-content-components
# Adding React state to markdown parser and save functinality

This PR helps to convert the react state to markdown and send the data to edx-platform. This is required to show or have a markdown format of the  Problems.

~~**Discussions**: Link to any public dicussions about this PR or the design/architecture. Otherwise omit this.~~

~~**Dependencies**: None~~

~~**Screenshots**: Always include screenshots if there is any change to the UI.~~

~~**Sandbox URL**: TBD - sandbox is being provisioned.~~

~~**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.~~

**Testing instructions**:

1. Set up the devstack as mentioned [here](https://github.com/edx/frontend-lib-content-components/pull/110#issue-1360810689)
2. There are tests written for the react state parser which you can have a look and run using `npm test -- src/editors/containers/ProblemEditor/data/ReactStateParser.test.js --collectCoverage=false`
3. From studio if we are entering to the MFE and if any change is made and saved, this just reflects in the markdown and not in OLX so xblock preview doesn't have any change.
4. Only when you disable the flags and click edit you can see the change in the markdown editor.
5. You can also put a `console.log(content)` [here](https://github.com/open-craft/frontend-lib-content-components/pull/2/files#diff-c2d3d9d3c28ca1b4aa77706aa0aefb06b4d2f0facbb3a90e96a67963ed656eb3R122) for Gallery and similar place in api.js for live, you need to do a persist log here.

**Author notes and concerns**:

1. OLX has to be updated which will be done in the next tickets
2. We have still to find out how to parse and make the difference between question, description and label.

**Reviewers**
- [ ] @navinkarkera

Fixes: https://tasks.opencraft.com/browse/BB-6668
Co-authored-by: <>
Private-ref: <>
